File: rest-service/src/main/java/org/veupathdb/service/multiblast/model/blast/BlastnConfig.java
package org.veupathdb.service.multiblast.model.blast;

import java.io.File;

import org.veupathdb.service.multiblast.model.CLISerializable;
import org.veupathdb.service.multiblast.model.ErrorMap;
import org.veupathdb.service.multiblast.model.Validatable;
import org.veupathdb.service.multiblast.service.jobs.BlastnValidator;

public class BlastnConfig extends StdBlastConfig implements CLISerializable, Validatable
{
  private QueryStrand   strand;
  private BlastnTask    task;
  private Integer       gapOpen;
  private Integer       gapExtend;
  private Integer       penalty;
  private Integer       reward;
  private Boolean       useIndex;
  private String        indexName;
  private Dust          dust;
  private File          filteringDb;
  private Integer       windowMaskerTaxID;
  private File          windowMaskerDB;
  private Double        percIdentity;
  private Integer       cullingLimit;
  private Double        bestHitOverhang;
  private Double        bestHitScoreEdge;
  private Boolean       subjectBestHit;
  private TemplateType  templateType;
  private Integer       templateLength;
  private Boolean       sumStats;

  @Override
  public ErrorMap validate() {
    return BlastnValidator.validate(this);
  }

  public QueryStrand getStrand() {
    return strand;
  }

  public void setStrand(QueryStrand strand) {
    this.strand = strand;
  }

  public BlastnTask getTask() {
    return task;
  }

  public void setTask(BlastnTask task) {
    this.task = task;
  }

  public Integer getGapOpen() {
    return gapOpen;
  }

  public void setGapOpen(Integer gapOpen) {
    this.gapOpen = gapOpen;
  }

  public Integer getGapExtend() {
    return gapExtend;
  }

  public void setGapExtend(Integer gapExtend) {
    this.gapExtend = gapExtend;
  }

  public Integer getPenalty() {
    return penalty;
  }

  public void setPenalty(Integer penalty) {
    this.penalty = penalty;
  }

  public Integer getReward() {
    return reward;
  }

  public void setReward(Integer reward) {
    this.reward = reward;
  }

  public Boolean getUseIndex() {
    return useIndex;
  }

  public void setUseIndex(Boolean useIndex) {
    this.useIndex = useIndex;
  }

  public String getIndexName() {
    return indexName;
  }

  public void setIndexName(String indexName) {
    this.indexName = indexName;
  }

  public Dust getDust() {
    return dust;
  }

  public void setDust(Dust dust) {
    this.dust = dust;
  }

  public File getFilteringDb() {
    return filteringDb;
  }

  public void setFilteringDb(File filteringDb) {
    this.filteringDb = filteringDb;
  }

  public Integer getWindowMaskerTaxID() {
    return windowMaskerTaxID;
  }

  public void setWindowMaskerTaxID(Integer windowMaskerTaxID) {
    this.windowMaskerTaxID = windowMaskerTaxID;
  }

  public File getWindowMaskerDB() {
    return windowMaskerDB;
  }

  public void setWindowMaskerDB(File windowMaskerDB) {
    this.windowMaskerDB = windowMaskerDB;
  }

  public Double getPercIdentity() {
    return percIdentity;
  }

  public void setPercIdentity(Double percIdentity) {
    this.percIdentity = percIdentity;
  }

  public Integer getCullingLimit() {
    return cullingLimit;
  }

  public void setCullingLimit(Integer cullingLimit) {
    this.cullingLimit = cullingLimit;
  }

  public Double getBestHitOverhang() {
    return bestHitOverhang;
  }

  public void setBestHitOverhang(Double bestHitOverhang) {
    this.bestHitOverhang = bestHitOverhang;
  }

  public Double getBestHitScoreEdge() {
    return bestHitScoreEdge;
  }

  public void setBestHitScoreEdge(Double bestHitScoreEdge) {
    this.bestHitScoreEdge = bestHitScoreEdge;
  }

  public Boolean getSubjectBestHit() {
    return subjectBestHit;
  }

  public void setSubjectBestHit(Boolean subjectBestHit) {
    this.subjectBestHit = subjectBestHit;
  }

  public TemplateType getTemplateType() {
    return templateType;
  }

  public void setTemplateType(TemplateType templateType) {
    this.templateType = templateType;
  }

  public Integer getTemplateLength() {
    return templateLength;
  }

  public void setTemplateLength(Integer templateLength) {
    this.templateLength = templateLength;
  }

  public Boolean getSumStats() {
    return sumStats;
  }

  public void setSumStats(Boolean sumStats) {
    this.sumStats = sumStats;
  }


}

File: lib-jvm-mblast-sdk/src/main/kotlin/org/veupathdb/lib/mblast/sdk/query/blast/BlastNConfig.kt
package org.veupathdb.lib.mblast.sdk.query.blast

import com.fasterxml.jackson.annotation.JsonGetter
import com.fasterxml.jackson.annotation.JsonSetter

class BlastNConfig : BlastQueryConfig() {

  override val tool: BlastQueryTool
    get() = BlastQueryTool.BlastN

  @get:JsonGetter("strand")
  @set:JsonSetter("strand")
  var strand: BlastStrand? = null

  @get:JsonGetter("task")
  @set:JsonSetter("task")
  var task: BlastNTask? = null

  @get:JsonGetter("wordSize")
  @set:JsonSetter("wordSize")
  var wordSize: UInt? = null

  @get:JsonGetter("gapOpen")
  @set:JsonSetter("gapOpen")
  var gapOpen: Int? = null

  @get:JsonGetter("gapExtend")
  @set:JsonSetter("gapExtend")
  var gapExtend: Int? = null

  @get:JsonGetter("penalty")
  @set:JsonSetter("penalty")
  var penalty: Int? = null

  @get:JsonGetter("reward")
  @set:JsonSetter("reward")
  var reward: UInt? = null

  @get:JsonGetter("useIndex")
  @set:JsonSetter("useIndex")
  var useIndex: Boolean? = null

  @get:JsonGetter("indexName")
  @set:JsonSetter("indexName")
  var indexName: String? = null

  @get:JsonGetter("dust")
  @set:JsonSetter("dust")
  var dust: BlastNDust? = null

  @get:JsonGetter("dbSoftMask")
  @set:JsonSetter("dbSoftMask")
  var dbSoftMask: String? = null

  @get:JsonGetter("dbHardMask")
  @set:JsonSetter("dbHardMask")
  var dbHardMask: String? = null

  @get:JsonGetter("percentIdentity")
  @set:JsonSetter("percentIdentity")
  var percentIdentity: Double? = null

  @get:JsonGetter("cullingLimit")
  @set:JsonSetter("cullingLimit")
  var cullingLimit: UInt? = null

  @get:JsonGetter("bestHitOverhang")
  @set:JsonSetter("bestHitOverhang")
  var bestHitOverhang: Double? = null

  @get:JsonGetter("bestHitScoreEdge")
  @set:JsonSetter("bestHitScoreEdge")
  var bestHitScoreEdge: Double? = null

  @get:JsonGetter("subjectBestHit")
  @set:JsonSetter("subjectBestHit")
  var subjectBestHit: Boolean? = null

  @get:JsonGetter("templateType")
  @set:JsonSetter("templateType")
  var templateType: BlastNTemplateType? = null

  @get:JsonGetter("templateLength")
  @set:JsonSetter("templateLength")
  var templateLength: BlastNTemplateLength? = null

  @get:JsonGetter("sumStats")
  @set:JsonSetter("sumStats")
  var sumStats: Boolean? = null

  @get:JsonGetter("xDropoffPrelimGapped")
  @set:JsonSetter("xDropoffPrelimGapped")
  var xDropoffPrelimGapped: Double? = null

  @get:JsonGetter("xDropoffFinalGapped")
  @set:JsonSetter("xDropoffFinalGapped")
  var xDropoffFinalGapped: Double? = null

  @get:JsonGetter("nonGreedy")
  @set:JsonSetter("nonGreedy")
  var nonGreedy: Boolean? = null

  @get:JsonGetter("minRawGappedScore")
  @set:JsonSetter("minRawGappedScore")
  var minRawGappedScore: Int? = null

  @get:JsonGetter("ungappedOnly")
  @set:JsonSetter("ungappedOnly")
  var ungappedOnly: Boolean? = null

  @get:JsonGetter("offDiagonalRange")
  @set:JsonSetter("offDiagonalRange")
  var offDiagonalRange: Int? = null
}
